addon/components/custom-subsidy-form-fields/climate-subsidy-costs-table/table-row-custom-data.js

```diff
@@ -0,0 +1,407 @@
+import Component from '@glimmer/component';
+import rdflib from 'browser-rdflib';
+import { v4 as uuidv4 } from 'uuid';
+import { tracked } from '@glimmer/tracking';
+import { A } from '@ember/array';
+import { action } from '@ember/object';
+import { guidFor } from '@ember/object/internals';
+import { scheduleOnce } from '@ember/runloop';
+
+import {
+  RDF,
+  XSD,
+  removeSimpleFormValue,
+} from '@lblod/submission-form-helpers';
+
+const MU = new rdflib.Namespace('http://mu.semte.ch/vocabularies/core/');
+
+const climateBaseUri = 'http://data.lblod.info/vocabularies/subsidie/climate/';
+const climateTableBaseUri = 'http://data.lblod.info/climate-tables';
+
+const tableEntryBaseUri = 'http://data.lblod.info/id/climate-table/row-entry';
+const ClimateEntryType = new rdflib.NamedNode(`${climateBaseUri}ClimateEntry`);
+const climateEntryPredicate = new rdflib.NamedNode(
+  `${climateBaseUri}climateEntry`
+);
+const climateEntryCustomData = new rdflib.NamedNode(
+  `${climateBaseUri}customData`
+);
+const actionDescriptionPredicate = new rdflib.NamedNode(
+  `${climateBaseUri}actionDescription`
+);
+const amountPerActionPredicate = new rdflib.NamedNode(
+  `${climateBaseUri}amountPerAction`
+);
+const restitutionPredicate = new rdflib.NamedNode(
+  `${climateBaseUri}restitution`
+);
+const hasInvalidRowPredicate = new rdflib.NamedNode(
+  `${climateTableBaseUri}/hasInvalidClimateTableEntry`
+);
+const customDescriptionPredicate = new rdflib.NamedNode(
+  `${climateBaseUri}customDescription`
+);
+const toRealiseUnitsPredicate = new rdflib.NamedNode(
+  `${climateBaseUri}toRealiseUnits`
+);
+const costPerUnitPredicate = new rdflib.NamedNode(
+  `${climateBaseUri}costPerUnit`
+);
+
+export default class CustomSubsidyFormFieldsClimateSubsidyCostsTableTableRowCustomDataComponent extends Component {
+  id = guidFor(this);
+  @tracked tableEntryUri = null;
+  @tracked description = '';
+  @tracked amount = null;
+  @tracked restitution = null;
+  @tracked toRealiseUnits = null;
+  @tracked costPerUnit = null;
+  @tracked descriptionErrors = [];
+  @tracked toRealiseUnitsErrors = [];
+  @tracked costPerUnitErrors = [];
+
+  get storeOptions() {
+    return this.args.storeOptions;
+  }
+
+  // Do we need this for custom data?
+  get businessRuleUri() {
+    return new rdflib.NamedNode(this.args.businessRuleUriStr);
+  }
+
+  get climateTableSubject() {
+    return this.args.climateTableSubject;
+  }
+
+  // TODO: this logic probably needs to be adjusted
+  get hasValues() {
+    const values = this.storeOptions.store.match(
```

Comment:
The plan is to pass the tableEntry uri from the parent component and use that uri to check for values. Is that ok?